camel-k-runtime-core/src/main/java/org/apache/camel/k/cutomizer/StreamCachingContextCustomizer.java

```diff
@@ -0,0 +1,124 @@
+package org.apache.camel.k.cutomizer;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.k.ContextCustomizer;
+import org.apache.camel.k.Runtime;
+import org.apache.camel.spi.StreamCachingStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StreamCachingContextCustomizer implements ContextCustomizer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StreamCachingContextCustomizer.class);
+
+    private boolean streamCachingEnabled;
+    private boolean streamCachingAnySpoolRules;
+    private int streamCachingBufferSize;
+    private boolean streamCachingRemoveSpoolDirectoryWhenStopping;
+    private String streamCachingSpoolChiper;
+    private String streamCachingSpoolDirectory;
+    private long streamCachingSpoolThreshold;
+    private String streamCachingSpoolUsedHeapMemoryLimit;
+    private int streamCachingSpoolUsedHeapMemoryThreshold;
+
+    public int getStreamCachingBufferSize() {
```

Comment:
maybe we can remove the StreamCaching from methods name so to customize it with properties we do not have duplication like

camel.customizer.streamcaching.bufferSize

vs

camel.cusimizer.streamcaching.streamcaching.streamCachingBufferSize

Reply:
We surely cane, and we did ;)